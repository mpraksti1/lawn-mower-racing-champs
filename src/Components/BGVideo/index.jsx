import React from 'react';
import PropTypes from 'prop-types';
import StyledVideo from './styles';

const BGVideo = (props) => {
  const { url } = props;

  return (
    <StyledVideo {...props}>
      <video autoPlay muted loop id="myVideo">
        <source src={url} type="video/mp4" />
      </video>
    </StyledVideo>
  );
};

BGVideo.defaultProps = {
  dark: false,
};

BGVideo.propTypes = {
  url: PropTypes.string.isRequired,
  dark: PropTypes.bool,
};

export default BGVideo;
