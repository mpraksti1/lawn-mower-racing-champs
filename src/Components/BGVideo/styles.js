import styled, { css } from 'styled-components';

const StyledVideo = styled.div`
  position: fixed;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  overflow: hidden;
  z-index: -100;

  @media (max-width: 767px) {
    background: url("/assets/images/mobile-bg.jpg") center center/cover no-repeat;
  }

  video {
    position: absolute;
    top: 50%;
    left: 50%;
    width: auto;
    height: auto;
    min-width: 100%;
    min-height: 100%;
    transform: translate(-50%, -50%);

    ${({ dark }) => dark && css`
      filter: brightness(0.4);
    `}

    ${({ blur }) => blur && css`
      filter: blur(7px);
    `}

    @media (max-width: 767px) {
      display: none;
    }
  }
`;

StyledVideo.displayName = 'VideoStyled';

export default StyledVideo;
