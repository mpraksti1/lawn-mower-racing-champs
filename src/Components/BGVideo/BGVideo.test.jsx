import BGVideo from '.';

describe('BG video component', () => {
  it('renders without crashing', () => {
    shallow(<BGVideo url="http://google.com" />);
  });

  it('matches snapshot', () => {
    const wrapper = shallow(<BGVideo url="http://google.com" />);

    expect(wrapper).toMatchSnapshot();
  });
});
