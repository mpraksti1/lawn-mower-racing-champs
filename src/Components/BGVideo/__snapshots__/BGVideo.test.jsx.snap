// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BG video component matches snapshot 1`] = `
<VideoStyled
  dark={false}
  url="http://google.com"
>
  <video
    autoPlay={true}
    id="myVideo"
    loop={true}
    muted={true}
  >
    <source
      src="http://google.com"
      type="video/mp4"
    />
  </video>
</VideoStyled>
`;
